src/client/network.rs

```diff
@@ -367,11 +375,18 @@ impl KafkaConnection {
     fn new(id: u32,
            host: &str,
            rw_timeout: Option<Duration>,
-           security: Option<SslConnector>)
+           security: Option<(SslConnector, bool)>)
            -> Result<KafkaConnection> {
         let stream = try!(TcpStream::connect(host));
         let stream = match security {
-            Some(ref connector) => KafkaStream::Ssl(try!(connector.clone().connect(host, stream))),
+            Some((connector, verify_server_name)) => {
+                let connection = if verify_server_name {
+                    try!(connector.connect(host, stream))
```

Comment:
@flier thank you for the investigation and this update commit!

indeed, i originally set up my local certificates with a `CN=localhost` and set up the local kafka broker to advertise itself under `localhost:9093`. so, it should have worked. however, you made me dig a little bit deeper and i think i've found the core problem. with the following modification in your original PR (i.e. the state of your branch preceding this commit):

```rust
...
        let stream = try!(TcpStream::connect(host));
        let stream = match security {
            Some(ref connector) => {
                let domain = match host.rfind(':') {
                    None => host,
                    Some(i) => &host[..i],
                };
                debug!("Attempting to establish a secured connection to {} (domain: {})", host, domain);
                KafkaStream::Ssl(try!(connector.clone().connect(domain, stream)))
            }
            None => KafkaStream::Plain(stream),
        };
...
```

it started to work as expected. the point is, that `host` as given to this method is actually `domain:port` - [see here](https://github.com/spicavigo/kafka-rust/blob/master/src/client/state.rs#L312); there's actually [a ticket for that](https://github.com/spicavigo/kafka-rust/issues/88) but it won't be easy to resolve it - and, thus, hostname verification naturally fails due to `"domain:port" != "domain"`.

apart from that, yes, i think having the possibility to turn it off hostname verification is what we do want to provide. and i warmly welcome this commit. feel free to use the code i pasted here to apply it to your commit.